clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefCallArgsChecker.cpp

```diff
@@ -332,6 +345,14 @@ class UncheckedCallArgsChecker final : public RawPtrRefCallArgsChecker {
     return isUncheckedPtr(QT);
   }
 
+  bool isSafePtr(const CXXRecordDecl *Record) const final {
+    return isRefCounted(Record) || isCheckedPtr(Record);
+  }
+
+  bool isSafePtrType(const QualType type) const final {
+    return isRefOrCheckedPtrType(type);
+  }
+
```

Comment:
Would it be possible to define these two methods in `RawPtrRefCallArgsChecker`?

Reply:
These are declared as pure virtual functions in `RawPtrRefCallArgsChecker`. Do you mean move one of the definitions to the base class so that some of these concrete subclasses won't have to define them??

Reply:
Ah, I didn't see they're pure virtual functions. Yeah, that was my intention, move them so we don't have to define them repeatedly.

Reply:
Are you fine with pure virtual in the base class? Or do you think we should move one of the implementation to the base?

Reply:
I'm fine with them being pure virtual in base class.